tools/jsonschema_to_rust_dto_enum_flags.py

```diff
@@ -0,0 +1,140 @@
+
+DEFAULT_INDENT = 4
+DOCS_URL = 'https://github.com/kamu-data/open-data-fabric/blob/master/open-data-fabric.md#{}-schema'
+ALLOWED_SCHEMAS = frozenset(['MetadataEvent'])
+
+PREAMBLE = f"""
+///////////////////////////////////////////////////////////////////////////////
+// WARNING: This file is auto-generated from Open Data Fabric Schemas
+// See: http://opendatafabric.org/
+///////////////////////////////////////////////////////////////////////////////
+
+use bitflags::bitflags;
+
+use crate::{{ {','.join(ALLOWED_SCHEMAS)} }};
+
+///////////////////////////////////////////////////////////////////////////////
+"""
+
+JsonType = dict[str, 'JsonType']
+
+
+def render(schemas_dir: str) -> None:
+    schemas = read_schemas(schemas_dir)
+
+    print(PREAMBLE)
+
+    for name in sorted(schemas.keys()):
+        if name not in ALLOWED_SCHEMAS:
+            continue
+
+        sch = schemas[name]
+
+        try:
+            if name == 'Manifest':
+                continue
+
+            print('/' * 80)
+            print(f'// {name}')
+            print('// ' + DOCS_URL.format(name.lower()))
+            print('/' * 80)
+            print()
+
+            for l in render_schema(name, sch):
+                print(l)
+            print()
+
+
+        except Exception as ex:
+            raise Exception(
+                f'Error while rendering {name} schema:\n{sch}'
+            ) from ex
+
+
+def read_schemas(schemas_dir: str) -> dict[str, JsonType]:
+    schemas = {}
+    read_schemas_rec(schemas_dir, schemas)
+    return schemas
+
+
+def read_schemas_rec(schemas_dir: str, schemas: dict[str, JsonType]) -> None:
+    for fname in os.listdir(schemas_dir):
+        path = os.path.join(schemas_dir, fname)
+
+        if os.path.isdir(path):
+            read_schemas_rec(path, schemas)
+            continue
+
+        with open(path) as f:
+            s = json.load(f)
+            fname = os.path.splitext(os.path.split(path)[-1])[0]
+            name = os.path.splitext(s['$id'].split('/')[-1])[0]
+            assert fname == name, f"{fname} != {name}"
+            schemas[name] = s
+
+
+def render_comment(text: str) -> Iterator[str]:
+    if not text:
+        return
+
+    for line in text.split('\n'):
+        yield f"/// {line.strip()}"
+
+
+def render_schema(name: str, sch: JsonType) -> Iterator[str]:
+    if 'oneOf' in sch:
+        yield from render_oneof(name, sch)
+    else:
+        raise Exception(f'Unsupported schema: {sch}')
+
+
+def render_oneof(name: str, sch: JsonType) -> Iterator[str]:
+    enum_name = f'{name}TypeFlags'
+
+    yield 'bitflags! {'
+    yield from indent(['#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]'])
+    yield from indent([f'pub struct {enum_name}: u32 {{'])
+
+    enames = []
+
+    for i, isch in enumerate(sch["oneOf"]):
+        ref = cast(str, isch["$ref"])
+        ename = ref.split('/')[-1]
+        yield from indent(indent([f'const {to_screaming_snake_case(ename)} = 1 << {i};']))
```

Comment:
What will happen to `const DATA_BLOCK = Self::ADD_DATA.bits() | Self::EXECUTE_TRANSFORM.bits();`?

Reply:
This has already been added via `dtos_extra.rs`:

```rust
impl MetadataEventTypeFlags {
    pub const DATA_BLOCK: Self =
        Self::from_bits_retain(Self::ADD_DATA.bits() | Self::EXECUTE_TRANSFORM.bits());
}
```
(c) https://github.com/kamu-data/kamu-cli/pull/543/files#diff-b967f0a7157d57be5b442d9f8b58c5828e05b3e2efd589e0ef8bf30228550fe9R305-R308